lib/rules/no-self-compare.js

```diff
@@ -12,6 +12,8 @@
 
 module.exports = {
     meta: {
+        type: "suggestion",
```

Comment:
This seems like it should be `problem` since it usually catches bugs.

Reply:
Agreed